core/eolearn/tests/test_utils/test_testing.py

```diff
@@ -0,0 +1,60 @@
+import datetime as dt
+
+import pytest
+
+from sentinelhub import CRS, BBox
+
+from eolearn.core.constants import FeatureType
+from eolearn.core.types import FeatureSpec, FeaturesSpecification
+from eolearn.core.utils.testing import patch_generator
+
+
+def test_patch_generator_default() -> None:
+    patch = patch_generator()
+    assert hasattr(patch, "bbox")
+    assert hasattr(patch, "timestamps")
+
+
+def test_patch_generator_not_default() -> None:
+    bbox = BBox((0, 0, 10, 10), crs=CRS("EPSG:32633"))
+    timestamps = [dt.datetime(2019, 1, 1)]
+    patch = patch_generator(bbox=bbox, timestamps=timestamps)
+
+    assert patch.bbox == bbox
+    assert patch.timestamps == timestamps
+
+
+@pytest.mark.parametrize(
+    "feature",
+    [
+        (FeatureType.DATA, "data"),
+        (FeatureType.MASK, "mask"),
+        (FeatureType.SCALAR, "scalar"),
+        (FeatureType.LABEL, "label"),
+        (FeatureType.DATA_TIMELESS, "data_timeless"),
+        (FeatureType.MASK_TIMELESS, "mask_timeless"),
+        (FeatureType.SCALAR_TIMELESS, "scalar_timeless"),
+        (FeatureType.LABEL_TIMELESS, "label_timeless"),
+    ],
+)
+def test_patch_generator_dim(feature: FeatureSpec) -> None:
+    patch = patch_generator(feature)
+    assert patch[feature].ndim == feature[0].ndim()
```

Comment:
this should be taken care of by the `EOPatch` (or rather by the `FeatureTypeDict` objects)
Did you find an example where the `ndim` of an feature value was wrong?

Reply:
yes if you throw FeatureType.SCALAR into patch_generator before u get 4 dim back this was the whole point of this "branch/issue/tests" ... but ATM agrre with your last point that for 2 dim data we do not know the size that we want

imo this test shuld stay mybe not in this form but for feturs that we know how to generate data one test and one that throw ValueError for all that we do not know how to generate ...

Reply:
really? please provide me with the code, because we clearly have a bug.

The 'regular' version fails as it should. So it's very weird that it goes through.
```python
In [1]: from eolearn.core import FeatureType, EOPatch

In [2]: import numpy as np

In [3]: eop = EOPatch()

In [4]: eop[FeatureType.SCALAR, "something"] = np.zeros((10, 10, 10, 10))
# THIS FAILS, AS IT SHOULD
```

Reply:
> yes if you throw FeatureType.SCALAR into patch_generator before u get 4 dim back

This was because we assumed in the code that "raster" means "image-type", which we then discovered it's not true, so we got height and width also for SCALAR* and LABEL*.  If the feature shape constructor works as is supposed to (now that we fixed it) this should be a problem anymore and there shouldn't be any bugs related to this. The only issue is our understanding of "is_raster" or perhaps it's definition.

Reply:
my point here is that the `assert` can never be triggered. L42 is unreachable, because if the data has the wrong `ndim` then L41 will throw an error.

Reply:
removed, add cases for fails